client/src/components/Form/Elements/FormFTP.vue

```diff
@@ -0,0 +1,193 @@
+                        :key="ftpFile.path"
+                        class="upload-ftp-row"
+                        @click="onRowClick(ftpFile)">
+                        <td class="ftp-name" colspan="4">
+                            <div
+                                class="icon"
+                                :class="{
+                                    [model.class_add]: !isSelected(ftpFile),
+                                    [model.class_remove]: isSelected(ftpFile),
+                                }"></div>
+                            {{ ftpFile.path }}
+                        </td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+        <div v-show="ftpFiles.length === 0" class="upload-ftp-warning warningmessage">no files.</div>
+    </div>
+</template>
+
+<script>
+import _ from "underscore";
+import $ from "jquery";
+import { getGalaxyInstance } from "app";
+import Utils from "utils/utils";
+import UploadUtils from "mvc/upload/upload-utils";
+
+export default {
+    props: ["options"],
+    data() {
+        return {
+            model: {
+                cls: "upload-ftp",
+                class_add: "class_add",
+                class_remove: "class_remove",
+                class_partial: "class_partial",
+                help_enabled: true,
+                oidc_text: `<br/>If you are signed-in to Galaxy using a third-party identity and you <strong>do not have a Galaxy password</strong> please use the reset password option in the login form with your email to create a password for your account.`,
+                help_text: `This Galaxy server allows you to upload files via FTP. To upload some files, log in to the FTP server at <strong>${options.ftp_upload_site}</strong> using your Galaxy credentials. For help visit the <a href="https://galaxyproject.org/ftp-upload/" target="_blank">tutorial</a>.`,
+                collection: null,
+                onchange: () => {},
+                onadd: () => {},
+                onremove: () => {},
+            },
+            ftpFiles: [],
+            waiting: true,
+            totalSize: 0,
+            isAllSelected: false,
+            ftpIndex: {},
+        };
+    },
+    created() {
+        this.model = _.extend(this.model, this.options);
+        this.collection = this.model.collection;
+        const Galaxy = getGalaxyInstance();
+        if (Galaxy.config.enable_oidc) {
+            this.model.help_text = this.model.help_text + this.model.oidc_text;
+        }
+    },
+    mounted() {
+        UploadUtils.getRemoteFiles(
+            (ftp_files) => {
+                this.ftpFiles = ftp_files;
+                this._index();
+                this._renderTable();
+            },
+            () => {
+                this._renderTable();
+            }
+        );
+    },
+    methods: {
+        /** Fill table with ftp entries */
+        _renderTable: function () {
```

Comment:
Please break this down and generate the data for a table to be injected into a standard vue table component.

Reply:
@guerler @ElectronicBlueberry  For this change request on my PR, would appreciate some guidance via [this QA Branch's PR with instructions](https://github.com/galaxyproject/galaxy/pull/16281) (not for merge, just to locally QA the FTP component).